app/src/main/java/org/oppia/android/app/devoptions/platformparameters/PlatformParameterItemViewModel.kt

```diff
@@ -0,0 +1,101 @@
+package org.oppia.android.app.devoptions.platformparameters
+
+import androidx.annotation.ColorInt
+import androidx.databinding.ObservableField
+import org.oppia.android.app.model.PlatformParameterId
+import org.oppia.android.app.model.PlatformParameterValue
+import org.oppia.android.app.model.SyncStatus
+import org.oppia.android.app.translation.AppLanguageResourceHandler
+import org.oppia.android.app.view.models.R
+import org.oppia.android.app.viewmodel.ObservableViewModel
+import org.oppia.android.util.locale.OppiaLocale
+
+/**
+ * [ViewModel] for displaying a platform parameter in the RecyclerView of
+ * [PlatformParametersFragment].
+ */
+class PlatformParameterItemViewModel(
+  val platformParameterId: PlatformParameterId,
+  val currentValue: PlatformParameterValue,
+  val syncStatus: SyncStatus,
+  private val machineLocale: OppiaLocale.MachineLocale,
+  private val resourceHandler: AppLanguageResourceHandler
+) : ObservableViewModel() {
+  /** The observable boolean value of the parameter, used for switch toggles in UI. */
+  val isChecked = ObservableField(currentValue.boolean)
+
+  /** Whether the input field should be visible (i.e., if the value is not a boolean). */
+  val isInputVisible = ObservableField(!currentValue.hasBoolean())
```

Comment:
The term input visible being a deciding variable could refer to both types. Since the views are determined by it being an input text - so suggesting this.
```suggestion
  val isTextInputMode = ObservableField(!currentValue.hasBoolean())
```

Reply:
updated.